api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +563,122 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      Preconditions.checkNotNull(crs, "CRS cannot be null");
+      this.crs = crs;
+    }
+
+    public static GeometryType get() {
+      return new GeometryType("");
+    }
+
+    public static GeometryType of(String crs) {
+      return new GeometryType(crs == null ? "" : crs);
+    }
+
+    @Override
+    public TypeID typeId() {
+      return TypeID.GEOMETRY;
+    }
+
+    public String crs() {
+      return crs;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      } else if (!(o instanceof GeometryType)) {
+        return false;
+      }
+
+      GeometryType that = (GeometryType) o;
+      return crs.equals(that.crs);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(GeometryType.class, crs);
+    }
+
+    @Override
+    public String toString() {
+      return String.format("geometry(%s)", crs);
```

Comment:
Using empty string for CRS would produce `geometry()` instead of `geometry`. I would also not use `geometry(crs84)` when that is equivalent to `geometry`. So whatever we store for the default CRS (null?) this should translate the default to `geometry`.

Reply:
The default CRS is stored as `null`, it will be translated to `geometry` for such cases.

Reply:
Looks good. The only remaining case is normalizing `OGC:CRS84`, which I suggested in the constructor comment above.